src/pip/_internal/cli/req_command.py

```diff
@@ -268,11 +277,37 @@ def get_requirements(
                 )
                 requirements.append(req_to_add)
 
+        if options.scripts:
+            if len(options.scripts) > 1:
+                raise CommandError("--script can only be given once")
+
+            script = options.scripts[0]
+            script_metadata = pep723_metadata(script)
+
+            script_requires_python = script_metadata.get("requires-python", "")
+
+            if script_requires_python and not options.ignore_requires_python:
+                target_python = make_target_python(options)
+
+                if not check_requires_python(
+                    requires_python=script_requires_python,
+                    version_info=target_python.py_version_info,
+                ):
+                    raise UnsupportedPythonVersion(
+                        f"Script {script!r} requires a different Python: "
+                        f"{target_python.py_version} not in {script_requires_python!r}"
+                    )
+
+            for req in script_metadata.get("dependencies", []):
+                requirements.append(
```

Comment:
`ruff` is upset that I wrote a loop with `.append()` here (`PERF401`) but I don't know if a comprehension would really be justified here? I can maybe see the case for `requirements.extend(...)` but it still seems like it'd be harder to read to me. Looking for feedback.